src/api/v2/orders.rs

```diff
@@ -1,13 +1,35 @@
 // Copyright (C) 2019-2022 The apca Developers
 // SPDX-License-Identifier: GPL-3.0-or-later
 
-use serde::Serialize;
 use serde_urlencoded::to_string as to_query;
+use serde::Serialize;
+use serde::Serializer;
 
 use crate::api::v2::order::Order;
 use crate::Str;
 
 
+/// Serialize a slice into a string of textual representations of the
+/// elements separated by comma.
+fn slice_to_str<S>(slice: &[String], serializer: S) -> Result<S::Ok, S::Error>
+where
+  S: Serializer,
+{
+  if !slice.is_empty() {
+    // `serde_urlencoded` seemingly does not know how to handle a
+    // `Vec`. So what we do is we convert each and every element to a
+    // string and then concatenate them, separating each by comma.
+    let s = slice
+      .iter()
+      .map(|val| val.to_uppercase())
+      .collect::<Vec<_>>()
+      .join(",");
+    serializer.serialize_str(&s)
+  } else {
+    serializer.serialize_none()
+  }
+}
+
```

Comment:
I just found and pushed https://github.com/d-e-s-o/apca/tree/topic/news. I suggest extracting the logic generalizing the existing function from https://github.com/d-e-s-o/apca/commit/855430275fc5f6dfd5455b2c7e0684b46f3e5d99 and https://github.com/d-e-s-o/apca/commit/818604498376726254c2deacc1af4116e9d373b3 to reduce duplication. Note sure if we actually need to upper case, though? If these are symbols we probably shouldn't mess with them at this stage.

(also, this branch actually contains the news API plumbing already, I see, which was one of the TODOs I mentioned)

Reply:
I see - this makes sense. I am a little unclear as to how you'd like me to proceed though.

Should I build on top of the topic/news branch? Or should I wait for that branch to be pushed and merged into devel, then merge devel into my branch and build on top of it from there? Or should I copy/paste the code into this branch and then you would merge into yours? 

Thanks for the help - I am fairly new when it comes to best practices for open source contributions.

Reply:
Ah, nah, that was just something I had lying around so you are free to take what you need from it and leave the rest. You can just check out the branch and then split the commits however you like. I don't know if that's best practice, but it works :-) (though in general, you probably shouldn't be taking random code snippets from elsewhere unless you have verified compatible licenses or without permission because of copyright)

Anyway, let me do that quickly as I authored the original changes anyway. I'll push these changes to `devel` and then you can use them.

Reply:
Perfect, thank you.

Reply:
Okay, I just pushed it to `devel` as 9954c9ffd2f46f79b77de061de44852ffe8ce993, so you should be able to use it from there once you rebase.

Reply:
Great, thank you. FYI - I updated the name_fn to return a String instead of &str since other serialization methods of this sort, like mine, will most likely be dealing with Strings. Let me know if there is a different/better way to do this.

Reply:
I think it's fine. Though, it would be better to use a [`Cow`](https://doc.rust-lang.org/std/borrow/enum.Cow.html) to still keep the ability to work with static strings and not having to needlessly allocate. I've taken care of that already, as it's a tiny change only.